### PR TITLE
fix(images): Fix image issues, support MDX

### LINKED ISSUE #16: Fix image issue
Images don't load properly in the build.
Can consider moving to a different MD file reader that uses next/images

### PR DESCRIPTION
**Pull request by ryanntannn**
## Description
- Add next-mdx-remote
- Remove markdown-it
- Add global alias for image loading
- Use next-images by default for MDX images
- Fix background image source

## Expected behaviour
- Images load properly and consistently

## Current behaviour
- Post images loading inconsistently from external sources
- Background image fails to load after refreshing in non-index page.

## Steps to test
1. `yarn build`
2. `yarn start`
3. make sure images are working in blog and events posts

## Linked PRs / Issue

fixes #16

### CODE REVIEW

File: features/file-read/mdx-components.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import Image, { ImageProps } from "next/image";
+
+const ResponsiveImage = (props: ImageProps) => (
+  // Alt text will be in props
+  // eslint-disable-next-line jsx-a11y/alt-text
+  <Image layout="responsive" width={0} height={0} {...props} />
```
Comment (natashaannn) [inline features/file-read/mdx-components.tsx:6]: I think `layout` is deprecated; not that it affects the code much, but can just use `responsive` instead: https://nextjs.org/docs/api-reference/next/legacy/image

File: pages/events/[slug].tsx [hunk 2]
```diff
@@ -20,13 +20,17 @@ import {
   getFileReadStaticPaths,
   getFileReadStaticProps,
 } from "../../features/file-read";
+import { MDXRemote } from "next-mdx-remote";
+import { components } from "../../features/file-read/mdx-components";
 
-export type EventPostPageProps = { frontmatter: EventData; content: string };
+export type EventPostPageProps = { frontmatter: EventData; content: any };
```
Comment (natashaannn) [inline pages/events/[slug].tsx:26]: `content` type here should be `MDXRemoteSerializeResult` aye?
https://github.com/hashicorp/next-mdx-remote

**Review by natashaannn — CHANGES_REQUESTED** [review]
Just some small type changes!

File: features/file-read/mdx-components.tsx (hunk 1 above)
Comment (ryanntannn) [reply features/file-read/mdx-components.tsx]: completely removed the Image override as i realized we don't really need it

**Pull request merged**